parastell/magnet_coils.py

```diff
@@ -436,6 +409,87 @@ def create_magnet(self):
+
+        Arguments:
+            lower_bound (float): lower bound of toroidal extent [rad].
+            upper_bound (float): upper bound of toroidal extent [rad].
+
+        Returns:
+            in_toroidal_extent (bool): flag to indicate whether coil lies
+                within toroidal bounds.
+        """
+        # Compute toroidal angle of each point in filament
+        toroidal_angles = np.arctan2(self._coords[:, 1], self._coords[:, 0])
+        # Ensure angles are positive
+        toroidal_angles = (toroidal_angles + 2 * np.pi) % (2 * np.pi)
+        # Compute bounds of toroidal extent of filament
+        min_tor_ang = np.min(toroidal_angles)
+        max_tor_ang = np.max(toroidal_angles)
+
+        # Determine if filament toroidal extent overlaps with that of model
+        if (min_tor_ang >= lower_bound or min_tor_ang <= upper_bound) or (
+            max_tor_ang >= lower_bound or max_tor_ang <= upper_bound
+        ):
+            in_toroidal_extent = True
+        else:
+            in_toroidal_extent = False
+
+        return in_toroidal_extent
+
+    def com_toroidal_angle(self):
+        """Computes the toroidal angle of the coil center of mass, based on
+        filament coordinates.
+
+        Returns:
+            (float): toroidal angle of coil center of mass [rad].
+        """
+        return np.arctan2(self.center_of_mass[1], self.center_of_mass[0])
+
+    def get_ob_mp_index(self):
+        """Finds the index of the outboard midplane coordinate on a coil
+        filament.
+
+        Returns:
+            outboard_index (int): index of the outboard midplane point.
+        """
+        # Compute radial distance of coordinates from z-axis
+        radii = np.linalg.norm(self.coords[:, :2], axis=1)
+        # Determine whether adjacent points cross the midplane (if so, they will
+        # have opposite signs)
+        midplane_flags = -np.sign(
+            self.coords[:, 2]
+            * np.append(self.coords[1:, 2], self.coords[1, 2])
+        )
+        # Find index of outboard midplane point
+        outboard_index = np.argmax(midplane_flags * radii)
+
+        return outboard_index
+
+    def reorder_coords(self, index):
+        """Reorders coil filament coordinate loop about a given index.
+
+        Arguments:
+            index (int): index about which to reorder coordinate loop.
+        """
+        self.coords = reorder_loop(self.coords, index)
+
+    def orient_coords(self, positive=True):
+        """Orients coil filament coordinate loop such that they initially
+        progress positively or negatively.
+
+        Arguments:
+            positive (bool): progress coordinates in positive direciton
+                (defaults to True). If negative, coordinates will progress in
+                negative direction.
+        """
+        if (positive and self.coords[0, 2] > self.coords[1, 2]) or (
+            not positive and self.coords[0, 2] < self.coords[1, 2]
+        ):
```

Comment:
Since `positive` is a boolean, can this be just:
```suggestion
        if (positive == (self.coords[0, 2] > self.coords[1, 2]) ):
```

Reply:
Very nice